### PR TITLE
feature/streamp 1731 emit event on producer update

### PR DESCRIPTION
**Pull request by neraj9**
# stream-registry PR

Emit an event on Producer update

### Added
* A listener for producer events
* A Kafka handler for producer events

# PR Checklist Forms

- [x] CHANGELOG.md updated
- [ ] Reviewer assigned
- [ ] PR assigned (presumably to submitter)
- [ ] Labels added (enhancement, bug, documentation)

### CODE REVIEW

File: notification-support/src/main/java/com/expediagroup/streamplatform/streamregistry/core/events/ProducerNotificationEventUtils.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    val zone = key.getZone();
+
+    var zoneKey = AvroKey.newBuilder()
+        .setId(zone)
+        .setType(AvroKeyType.ZONE)
+        .build();
+
+    var domainKey = AvroKey.newBuilder()
+        .setId(streamDomain)
+        .setParent(zoneKey)
```
Comment (nahguam) [inline notification-support/src/main/java/com/expediagroup/streamplatform/streamregistry/core/events/ProducerNotificationEventUtils.java:52]: Parent of domain is `null`

File: notification-support/src/main/java/com/expediagroup/streamplatform/streamregistry/core/events/StreamBindingNotificationEventUtils.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+    var infrastructureKey = AvroKey.newBuilder()
+        .setId(infrastructureName)
+        .setParent(zoneKey)
+        .setType(AvroKeyType.INFRASTRUCTURE)
+        .build();
+
+    var domainKey = AvroKey.newBuilder()
+        .setId(streamDomain)
+        .setParent(infrastructureKey)
```
Comment (nahguam) [inline notification-support/src/main/java/com/expediagroup/streamplatform/streamregistry/core/events/StreamBindingNotificationEventUtils.java:59]: Null

File: notification-support/src/main/java/com/expediagroup/streamplatform/streamregistry/core/events/StreamBindingNotificationEventUtils.java [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+    var streamKey = AvroKey.newBuilder()
+        .setId(streamName)
+        .setParent(domainKey)
+        .setType(AvroKeyType.STREAM)
+        .build();
+
+    return AvroKey.newBuilder()
+        .setId(streamVersion.toString())
+        .setParent(streamKey)
```
Comment (nahguam) [inline notification-support/src/main/java/com/expediagroup/streamplatform/streamregistry/core/events/StreamBindingNotificationEventUtils.java:71]: Hmm how to set the parent when it has lineage from two sides (stream and infrastructure)

File: notification-support/src/main/java/com/expediagroup/streamplatform/streamregistry/core/events/ProducerNotificationEventUtils.java (hunk 1 above)
Comment (neraj9) [reply notification-support/src/main/java/com/expediagroup/streamplatform/streamregistry/core/events/ProducerNotificationEventUtils.java]: done

File: notification-support/src/main/java/com/expediagroup/streamplatform/streamregistry/core/events/StreamBindingNotificationEventUtils.java (hunk 3 above)
Comment (neraj9) [reply notification-support/src/main/java/com/expediagroup/streamplatform/streamregistry/core/events/StreamBindingNotificationEventUtils.java:71]: done

File: notification-support/src/main/java/com/expediagroup/streamplatform/streamregistry/core/events/StreamBindingNotificationEventUtils.java (hunk 2 above)
Comment (neraj9) [reply notification-support/src/main/java/com/expediagroup/streamplatform/streamregistry/core/events/StreamBindingNotificationEventUtils.java]: done

File: notification-support/src/main/java/com/expediagroup/streamplatform/streamregistry/core/events/StreamBindingNotificationEventUtils.java (hunk 3 above)
Comment (neraj9) [reply notification-support/src/main/java/com/expediagroup/streamplatform/streamregistry/core/events/StreamBindingNotificationEventUtils.java:71]: done

File: notification-support/src/main/java/com/expediagroup/streamplatform/streamregistry/core/events/StreamBindingNotificationEventUtils.java [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+    var streamKey = AvroKey.newBuilder()
+        .setId(streamName)
+        .setParent(domainKey)
+        .setType(AvroKeyType.STREAM)
+        .build();
+
+    var zoneKey = AvroKey.newBuilder()
+        .setId(infrastructureZone)
+        .setParent(domainKey)
```
Comment (nahguam) [inline notification-support/src/main/java/com/expediagroup/streamplatform/streamregistry/core/events/StreamBindingNotificationEventUtils.java:59]: zone's parent is `null`

File: notification-support/src/main/java/com/expediagroup/streamplatform/streamregistry/core/events/ProducerNotificationEventUtils.java [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+    val zone = key.getZone();
+
+    var domainKey = AvroKey.newBuilder()
+        .setId(streamDomain)
+        .setType(AvroKeyType.DOMAIN)
+        .build();
+
+    var zoneKey = AvroKey.newBuilder()
+        .setId(zone)
+        .setParent(domainKey)
```
Comment (nahguam) [inline notification-support/src/main/java/com/expediagroup/streamplatform/streamregistry/core/events/ProducerNotificationEventUtils.java:52]: Zone's parent is `null`

File: notification-support/src/main/java/com/expediagroup/streamplatform/streamregistry/core/events/ProducerNotificationEventUtils.java [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+    var streamVersionKey = AvroKey.newBuilder()
+        .setId(streamVersion.toString())
+        .setParent(streamKey)
+        .setType(AvroKeyType.STREAM_VERSION)
+        .build();
+
+    return AvroKey.newBuilder()
+        .setId(producerName)
+        .setParent(streamVersionKey)
+        .setType(AvroKeyType.PRODUCER)
```
Comment (nahguam) [inline notification-support/src/main/java/com/expediagroup/streamplatform/streamregistry/core/events/ProducerNotificationEventUtils.java:71]: now that we have lineage from both sides you could set the physical for a producer to the zone?

**Pull request merged**